Index: A3-Antistasi-Greenfork/src/description.ext
===================================================================
--- A3-Antistasi-Greenfork.orig/src/description.ext	2020-06-03 01:05:13.729420062 +0000
+++ A3-Antistasi-Greenfork/src/description.ext	2020-06-07 15:00:23.172977879 +0000
@@ -91,7 +91,7 @@
         title = "Enable Server Membership";
         texts[] = {"Yes","No"};
         values[] = {1,0};
-        default = 1;
+        default = 0;
         //function = "BIS_fnc_paramDaytime"; // (Optional) [[Functions_Library_(Arma_3)|Function]] [[call]]ed when player joins, selected value is passed as an argument
         //isGlobal = 1; // (Optional) 1 to execute script / function locally for every player who joins, 0 to do it only on server
     };
@@ -108,7 +108,7 @@
         title = "Enable Teamkill Punish";
         values[] = {1,0};
         texts[] = {"Yes","No"};
-        default = 1;
+        default = 0;
         //file = "setViewDistance.sqf"; // (Optional) Script [[call]]ed when player joins, selected value is passed as an argument
     };
     class mRadius
