---
 src/healandrepair.sqf |    2 ++
 src/initVar.sqf       |    1 +
 2 files changed, 3 insertions(+)

--- a/src/healandrepair.sqf
+++ b/src/healandrepair.sqf
@@ -7,6 +7,8 @@
 	if (hayACEMedical) then
 		{
 		[_x, _x] call ace_medical_fnc_treatmentAdvanced_fullHeal;
+		[_x] call ace_medical_treatment_fnc_fullHealLocal;
+		[_x, false] call ace_medical_fnc_setUnconscious;
 		}
 	else
 		{
--- a/src/initVar.sqf
+++ b/src/initVar.sqf
@@ -741,6 +741,7 @@
 		"ACE_salineIV_250",
 		"ACE_surgicalKit",
 		"ACE_tourniquet",
+		"ACE_splint",
 		"ACE_adenosine",
 		"ACE_atropine"
 	]; publicVariable "aceAdvMedItems";
