Index: A3-Antistasi-Greenfork/src/healandrepair.sqf
===================================================================
--- A3-Antistasi-Greenfork.orig/src/healandrepair.sqf	2020-04-24 15:45:33.044864800 +0000
+++ A3-Antistasi-Greenfork/src/healandrepair.sqf	2020-04-24 15:49:59.623670822 +0000
@@ -7,6 +7,8 @@
 	if (hayACEMedical) then
 		{
 		[_x, _x] call ace_medical_fnc_treatmentAdvanced_fullHeal;
+		[_x] call ace_medical_treatment_fnc_fullHealLocal;
+		[_x, false] call ace_medical_fnc_setUnconscious;
 		}
 	else
 		{
Index: A3-Antistasi-Greenfork/src/initVar.sqf
===================================================================
--- A3-Antistasi-Greenfork.orig/src/initVar.sqf	2020-04-24 15:45:33.044864800 +0000
+++ A3-Antistasi-Greenfork/src/initVar.sqf	2020-04-24 15:49:25.799823979 +0000
@@ -741,6 +741,7 @@
 		"ACE_salineIV_250",
 		"ACE_surgicalKit",
 		"ACE_tourniquet",
+		"ACE_splint",
 		"ACE_adenosine",
 		"ACE_atropine"
 	]; publicVariable "aceAdvMedItems";
