Index: A3-Antistasi-Greenfork/src/CREATE/FIAinitBASES.sqf
===================================================================
--- A3-Antistasi-Greenfork.orig/src/CREATE/FIAinitBASES.sqf	2020-03-29 14:38:34.868127240 +0000
+++ A3-Antistasi-Greenfork/src/CREATE/FIAinitBASES.sqf	2020-03-29 15:08:45.053456129 +0000
@@ -21,7 +21,7 @@
 			}
 		};
 	};
-[_unit] call A3A_fnc_initRevive;
+if (!hayACEMedical) then {[_unit] call A3A_fnc_initRevive};
 
 _unit allowFleeing 0;
 _tipo = typeOf _unit;
Index: A3-Antistasi-Greenfork/src/REINF/FIAinit.sqf
===================================================================
--- A3-Antistasi-Greenfork.orig/src/REINF/FIAinit.sqf	2020-03-29 14:38:35.136124920 +0000
+++ A3-Antistasi-Greenfork/src/REINF/FIAinit.sqf	2020-03-29 15:08:23.563642170 +0000
@@ -2,7 +2,7 @@
 
 _unit = _this select 0;
 
-[_unit] call A3A_fnc_initRevive;
+if (!hayACEMedical) then {[_unit] call A3A_fnc_initRevive};
 _unit setVariable ["spawner",true,true];
 if (_unit == leader _unit) then {group _unit setVariable ["origen",respawnBuenos]};
 _unit allowFleeing 0;
