Index: A3-Antistasi-Greenfork/src/Scripts/UPSMON/COMMON/vehicles/fnc/UPSMON_GetNearestVehicles.sqf
===================================================================
--- A3-Antistasi-Greenfork.orig/src/Scripts/UPSMON/COMMON/vehicles/fnc/UPSMON_GetNearestVehicles.sqf	2020-05-18 13:12:04.373821253 +0000
+++ A3-Antistasi-Greenfork/src/Scripts/UPSMON/COMMON/vehicles/fnc/UPSMON_GetNearestVehicles.sqf	2020-05-18 13:14:37.256143970 +0000
@@ -11,7 +11,7 @@
 Returns:
 	Array vehicles [[vehicles,emptypositions]]
 ****************************************************************/
-private["_npc","_vehtypes","_distance","_types","_marker","_OCercanos","_emptypositions","_vehicles","_Cargocount","_Gunnercount","_Commandercount","_Drivercount","_isuav","_inzone"];	
+private["_npc","_vehtypes","_distance","_types","_marker","_OCercanos","_emptypositions","_vehicles","_Cargocount","_Gunnercount","_Commandercount","_Drivercount","_isuav","_inzone","_islocked","_isowned"];	
 					
 _npc = _this select 0;	
 _types = _this select 1;
@@ -40,6 +40,14 @@
 		_Gunnercount = _x call UPSMON_Emptyturret; 
 		_Drivercount = (_x) emptyPositions "Driver"; 
 		_Commandercount = (_x) emptyPositions "Commander"; 
+
+		_isowned = false;
+
+		// A3-Antistasi hotfix
+		_islocked = _x getVariable "duenyo";
+		_islocked = not isNil "_islocked";
+
+		if (((side _npc) == resistance) and _islocked) then {_isowned = true};
 		
 		_emptypositions = _Cargocount + _Gunnercount + _Commandercount + _Drivercount;
 		
@@ -66,7 +74,7 @@
 						{
 							If (_x getvariable ["UPSMON_GrpId",0] == 0) then
 							{
-								If (_inzone) then
+								If (_inzone and not _isowned) then
 								{
 									_vehicles pushback [_x,_emptypositions,_points];
 								};
