Index: A3-Antistasi-Greenfork/src/resourcecheck.sqf
===================================================================
--- A3-Antistasi-Greenfork.orig/src/resourcecheck.sqf	2020-06-02 20:08:26.503823793 +0000
+++ A3-Antistasi-Greenfork/src/resourcecheck.sqf	2020-06-02 20:53:54.327802499 +0000
@@ -5,12 +5,13 @@
 private ["_texto"];
 scriptName "resourcecheck";
 _cuentaSave = 3600;
+nextIncome = time + incomeTick;
 
 while {true} do
 	{
 	//sleep 600;//600
 	nextTick = time + 600;
-	waitUntil {sleep 15; time >= nextTick};
+	[nextTick] call A3A_fnc_sleepUntil;
 	if (isMultiplayer) then {waitUntil {sleep 10; isPlayer theBoss}};
 	_suppBoost = 1+ ({lados getVariable [_x,sideUnknown] == buenos} count puertos);
 	_recAddSDK = 25;//0
@@ -113,10 +114,12 @@
 	_updated = [] call A3A_fnc_arsenalManage;
 	if (_updated != "") then {_texto = format ["%1<br/>Arsenal Updated<br/><br/>%2",_texto,_updated]};
 	[petros,"taxRep",_texto] remoteExec ["A3A_fnc_commsMP",[buenos,civilian]];
-	_hrAddBLUFOR = _hrAddBLUFOR + (server getVariable "hr");
-	_recAddSDK = _recAddSDK + (server getVariable "resourcesFIA");
-	server setVariable ["hr",_hrAddBLUFOR,true];
-	server setVariable ["resourcesFIA",_recAddSDK,true];
+
+	if((time + 45) > nextIncome) then {
+		nextIncome = time + incomeTick;
+		[_hrAddBLUFOR, _recAddSDK] call A3A_fnc_resourceTaxes;
+	};
+
 	bombRuns = bombRuns + (({lados getVariable [_x,sideUnknown] == buenos} count aeropuertos) * 0.25);
 	[petros,"taxRep",_texto] remoteExec ["A3A_fnc_commsMP",[buenos,civilian]];
 	[] call A3A_fnc_economicsAI;
Index: A3-Antistasi-Greenfork/src/functions.hpp
===================================================================
--- A3-Antistasi-Greenfork.orig/src/functions.hpp	2020-06-02 20:08:29.621802459 +0000
+++ A3-Antistasi-Greenfork/src/functions.hpp	2020-06-02 20:46:58.162852476 +0000
@@ -255,5 +255,7 @@
 	{
         class refreshCargoActions {file="refreshCargoActions.sqf";};
         class getDisplayName {file="getDisplayName.sqf";};
+        class resourceTaxes {file="resourceTaxes.sqf";};
+        class sleepUntil {file="sleepUntil.sqf";};
 	}
 }
Index: A3-Antistasi-Greenfork/src/resourceTaxes.sqf
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ A3-Antistasi-Greenfork/src/resourceTaxes.sqf	2020-04-14 19:31:35.431277130 +0000
@@ -0,0 +1,60 @@
+private ["_hr","_resourcesFIA", "_popFIA", "_hrCap", "_cashCap"];
+params ["_hrDelta", "_cashDelta"];
+
+if ((isNil "_hrDelta") or (isNil "_cashDelta")) exitWith {};
+if ((floor _hrDelta == 0) and (floor _cashDelta == 0)) exitWith {};
+
+_hr = server getVariable "hr";
+_resourcesFIA = server getVariable "resourcesFIA";
+
+_popFIA = 0;
+{
+	private _data = server getVariable _x;
+	private _population = _data select 0;
+	private _prestige = _data select 3;
+	_popFIA = _popFIA + round (_population * (_prestige / 100));
+} forEach ciudades;
+_hrCap = (round (_popFIA / 10)) max 10;
+_cashCap = (_popFIA * 10) max 10000;
+diag_log format ["Greenfork: Caps: pop=%1 HR=%2 cash=%3", _popFIA, _hrCap, _cashCap];
+
+if(_hr + _hrDelta < _hrCap) then {
+	private _newHr = _hr + _hrDelta;
+	diag_log format ["Greenfork: Caps: HR: not capping %1 -> %2", _hr, _newHr];
+	_hr = _newHr;
+} else {
+	private _difference = _hr + _hrDelta - _hrCap;
+
+	if(_difference > 5) then {
+		private _newHr = round ((_hr + _hrCap) / 2);
+		diag_log format ["Greenfork: Caps: HR: drain down %1 -> %2", _hr, _newHr];
+		_hr = _newHr;
+	} else {
+		diag_log format ["Greenfork: Caps: HR: cap %1 -> %2", _hr, _hrCap];
+		_hr = _hrCap;
+	};
+};
+
+if(_resourcesFIA + _cashDelta < _cashCap) then {
+	private _newCash = round (_resourcesFIA + _cashDelta);
+	diag_log format ["Greenfork: Caps: Cash: not capping %1 -> %2", _resourcesFIA, _newCash];
+	_resourcesFIA = _newCash;
+} else {
+	private _difference = _resourcesFIA + _cashDelta - _cashCap;
+
+	if(_difference > 5000) then {
+		private _newCash = round ((_resourcesFIA + _cashCap) / 2);
+		diag_log format ["Greenfork: Caps: Cash: drain down %1 -> %2", _resourcesFIA, _newCash];
+		_resourcesFIA = _newCash;
+	} else {
+		diag_log format ["Greenfork: Caps: Cash: cap %1 -> %2", _resourcesFIA, _cashCap];
+		_resourcesFIA = _cashCap;
+	};
+};
+
+
+if (_hr < 0) then {_hr = 0};
+if (_resourcesFIA < 0) then {_resourcesFIA = 0};
+
+server setVariable ["hr", _hr, true];
+server setVariable ["resourcesFIA", _resourcesFIA, true];
Index: A3-Antistasi-Greenfork/src/description.ext
===================================================================
--- A3-Antistasi-Greenfork.orig/src/description.ext	2020-06-02 20:08:29.505803253 +0000
+++ A3-Antistasi-Greenfork/src/description.ext	2020-06-02 20:48:54.151228158 +0000
@@ -181,6 +181,12 @@
         texts[] = {"No","Yes"};
         default = 1;
     };
+    class minutesPerIncomeTick
+    {
+        title = "Minutes per income period";
+        values[] = {10, 20, 30, 40, 60};
+        default = 20;
+    };
 };
 
 class CfgIdentities
Index: A3-Antistasi-Greenfork/src/initServer.sqf
===================================================================
--- A3-Antistasi-Greenfork.orig/src/initServer.sqf	2020-06-02 20:08:29.506803246 +0000
+++ A3-Antistasi-Greenfork/src/initServer.sqf	2020-06-02 20:08:29.653802240 +0000
@@ -29,6 +29,7 @@
 civTraffic = (paramsArray select 11) / 4; publicVariable "civTraffic";
 memberDistance = paramsArray select 13; publicVariable "memberDistance";
 limitedFT = if (paramsArray select 14 == 1) then {true} else {false}; publicVariable "limitedFT";
+incomeTick = (paramsArray select 15) * 60; publicVariable "incomeTick";
 _nul = call compile preprocessFileLineNumbers "initVar.sqf";
 initVar = true; publicVariable "initVar";
 savingServer = true;
Index: A3-Antistasi-Greenfork/src/statSave/loadAccount.sqf
===================================================================
--- A3-Antistasi-Greenfork.orig/src/statSave/loadAccount.sqf	2020-06-02 19:37:24.055135384 +0000
+++ A3-Antistasi-Greenfork/src/statSave/loadAccount.sqf	2020-06-02 20:58:38.732581419 +0000
@@ -174,6 +174,7 @@
 	*/
 	["posHQ"] call fn_LoadStat;
 	["nextTick"] call fn_LoadStat;
+	["nextIncome", true] call fn_LoadStat;
 	["estaticas"] call fn_LoadStat;//tiene que ser el último para que el sleep del borrado del contenido no haga que despawneen
 
 
Index: A3-Antistasi-Greenfork/src/statSave/saveFuncs.sqf
===================================================================
--- A3-Antistasi-Greenfork.orig/src/statSave/saveFuncs.sqf	2020-06-02 19:37:24.055135384 +0000
+++ A3-Antistasi-Greenfork/src/statSave/saveFuncs.sqf	2020-06-02 20:58:10.885708209 +0000
@@ -18,8 +18,8 @@
 
 fn_LoadStat =
 {
-	private ["_varName","_varvalue"];
-	_varName = _this select 0;
+	params ["_varName", ["_allowUninitialized", false]];
+	private ["_varvalue"];
 	if (worldName == "Tanoa") then
 		{
 		_varValue = profileNameSpace getVariable (_varName + serverID + "WotP")
@@ -28,8 +28,8 @@
 		{
 		if (side group petros == independent) then {_varValue = profileNameSpace getVariable (_varName + serverID + "Antistasi" + worldName)} else {_varValue = profileNameSpace getVariable (_varName + serverID + "AntistasiB" + worldName)};
 		};
-	if(isNil "_varValue") exitWith {diag_log format ["Antistasi: Error en Persistent Load. La variable %1 no existe",_varname]};
-	[_varName,_varValue] call fn_SetStat;
+	if(isNil "_varValue" && !_allowUninitialized) exitWith {diag_log format ["Antistasi: Error en Persistent Load. La variable %1 no existe",_varname]};
+	[_varName,_varValue,_allowUninitialized] call fn_SetStat;
 };
 
 //===========================================================================
@@ -44,13 +44,12 @@
 	"prestigeNATO","prestigeCSAT", "hr","planesAAFcurrent","helisAAFcurrent","APCAAFcurrent","tanksAAFcurrent","armas","items","mochis","municion","fecha", "WitemsPlayer","prestigeOPFOR","prestigeBLUFOR","resourcesAAF","resourcesFIA","skillFIA"];
 */
 specialVarLoads =
-["puestosFIA","minas","estaticas","cuentaCA","antenas","mrkNATO","mrkSDK","prestigeNATO","prestigeCSAT","posHQ", "hr","armas","items","mochis","municion","fecha", "prestigeOPFOR","prestigeBLUFOR","resourcesFIA","skillFIA","distanciaSPWN","civPerc","maxUnits","destroyedCities","garrison","tasks","scorePlayer","rankPlayer","smallCAmrk","dinero","miembros","vehInGarage","destroyedBuildings","personalGarage","idlebases","idleassets","chopForest","weather","killZones","jna_dataList","controlesSDK","loadoutPlayer","mrkCSAT","nextTick","bombRuns","dificultad","gameMode","securedCities"];
+["puestosFIA","minas","estaticas","cuentaCA","antenas","mrkNATO","mrkSDK","prestigeNATO","prestigeCSAT","posHQ", "hr","armas","items","mochis","municion","fecha", "prestigeOPFOR","prestigeBLUFOR","resourcesFIA","skillFIA","distanciaSPWN","civPerc","maxUnits","destroyedCities","garrison","tasks","scorePlayer","rankPlayer","smallCAmrk","dinero","miembros","vehInGarage","destroyedBuildings","personalGarage","idlebases","idleassets","chopForest","weather","killZones","jna_dataList","controlesSDK","loadoutPlayer","mrkCSAT","nextTick","nextIncome","bombRuns","dificultad","gameMode","securedCities"];
 //THIS FUNCTIONS HANDLES HOW STATS ARE LOADED
 fn_SetStat =
 {
-	_varName = _this select 0;
-	_varValue = _this select 1;
-	if(isNil '_varValue') exitWith {};
+	params ["_varName", "_varValue", ["_allowUninitialized", false]];
+	if(isNil '_varValue' && !_allowUninitialized) exitWith {};
 	if(_varName in specialVarLoads) then
 	{
 		if(_varName == 'cuentaCA') then {cuentaCA = _varValue; publicVariable "cuentaCA"};
@@ -106,6 +105,7 @@
 			};
 		if(_varName == 'bombRuns') then {bombRuns = _varValue; publicVariable "bombRuns"};
 		if(_varName == 'nextTick') then {nextTick = time + _varValue};
+		if(_varName == 'nextIncome') then {nextIncome = time + _varValue};
 		if(_varName == 'miembros') then {miembros = +_varValue; publicVariable "miembros"};
 		if(_varName == 'smallCAmrk') then {smallCAmrk = +_varValue};
 		if(_varName == 'mrkNATO') then {{lados setVariable [_x,malos,true]} forEach _varValue;};
Index: A3-Antistasi-Greenfork/src/statSave/saveLoop.sqf
===================================================================
--- A3-Antistasi-Greenfork.orig/src/statSave/saveLoop.sqf	2020-06-02 19:37:24.055135384 +0000
+++ A3-Antistasi-Greenfork/src/statSave/saveLoop.sqf	2020-06-02 20:53:08.146987871 +0000
@@ -69,6 +69,7 @@
 	["chopForest", chopForest] call fn_SaveStat;
 	["maxUnits", maxUnits] call fn_SaveStat;
 	["nextTick", nextTick - time] call fn_SaveStat;
+	["nextIncome", nextIncome - time] call fn_SaveStat;
 	/*
 	["unlockedWeapons", unlockedWeapons] call fn_SaveStat;
 	["unlockedItems", unlockedItems] call fn_SaveStat;
Index: A3-Antistasi-Greenfork/src/sleepUntil.sqf
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ A3-Antistasi-Greenfork/src/sleepUntil.sqf	2020-06-02 21:53:00.648344300 +0000
@@ -0,0 +1,19 @@
+params ["_future", ["_precision", 0.5], ["_busyloop", 10]];
+
+if((_future - _precision) < time) exitWith {};
+
+waitUntil {
+	private _diff = _future - time;
+
+	if (_diff > (_busyloop + _precision)) then {
+		sleep (_diff / 2);
+	};
+
+	_diff = _future - time;
+	_diff < (_busyloop + _precision)
+};
+
+waitUntil {
+	sleep _precision;
+	time > _future;
+};
