Index: A3-Antistasi-Greenfork/src/resourcecheck.sqf
===================================================================
--- A3-Antistasi-Greenfork.orig/src/resourcecheck.sqf	2020-05-19 16:20:26.853070018 +0000
+++ A3-Antistasi-Greenfork/src/resourcecheck.sqf	2020-05-21 06:00:21.987135617 +0000
@@ -9,7 +9,7 @@
 while {true} do
 	{
 	//sleep 600;//600
-	nextTick = time + 600;
+	nextTick = time + incomeTick;
 	waitUntil {sleep 15; time >= nextTick};
 	if (isMultiplayer) then {waitUntil {sleep 10; isPlayer theBoss}};
 	_suppBoost = 1+ ({lados getVariable [_x,sideUnknown] == buenos} count puertos);
@@ -113,10 +113,9 @@
 	_updated = [] call A3A_fnc_arsenalManage;
 	if (_updated != "") then {_texto = format ["%1<br/>Arsenal Updated<br/><br/>%2",_texto,_updated]};
 	[petros,"taxRep",_texto] remoteExec ["A3A_fnc_commsMP",[buenos,civilian]];
-	_hrAddBLUFOR = _hrAddBLUFOR + (server getVariable "hr");
-	_recAddSDK = _recAddSDK + (server getVariable "resourcesFIA");
-	server setVariable ["hr",_hrAddBLUFOR,true];
-	server setVariable ["resourcesFIA",_recAddSDK,true];
+
+	[_hrAddBLUFOR, _recAddSDK] call A3A_fnc_resourceTaxes;
+
 	bombRuns = bombRuns + (({lados getVariable [_x,sideUnknown] == buenos} count aeropuertos) * 0.25);
 	[petros,"taxRep",_texto] remoteExec ["A3A_fnc_commsMP",[buenos,civilian]];
 	[] call A3A_fnc_economicsAI;
Index: A3-Antistasi-Greenfork/src/functions.hpp
===================================================================
--- A3-Antistasi-Greenfork.orig/src/functions.hpp	2020-04-14 22:21:48.493360907 +0000
+++ A3-Antistasi-Greenfork/src/functions.hpp	2020-04-14 22:21:48.526360748 +0000
@@ -256,5 +256,6 @@
 	{
         class refreshCargoActions {file="refreshCargoActions.sqf";};
         class getDisplayName {file="getDisplayName.sqf";};
+        class resourceTaxes {file="resourceTaxes.sqf";};
 	}
 }
Index: A3-Antistasi-Greenfork/src/resourceTaxes.sqf
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ A3-Antistasi-Greenfork/src/resourceTaxes.sqf	2020-04-14 19:31:35.431277130 +0000
@@ -0,0 +1,60 @@
+private ["_hr","_resourcesFIA", "_popFIA", "_hrCap", "_cashCap"];
+params ["_hrDelta", "_cashDelta"];
+
+if ((isNil "_hrDelta") or (isNil "_cashDelta")) exitWith {};
+if ((floor _hrDelta == 0) and (floor _cashDelta == 0)) exitWith {};
+
+_hr = server getVariable "hr";
+_resourcesFIA = server getVariable "resourcesFIA";
+
+_popFIA = 0;
+{
+	private _data = server getVariable _x;
+	private _population = _data select 0;
+	private _prestige = _data select 3;
+	_popFIA = _popFIA + round (_population * (_prestige / 100));
+} forEach ciudades;
+_hrCap = (round (_popFIA / 10)) max 10;
+_cashCap = (_popFIA * 10) max 10000;
+diag_log format ["Greenfork: Caps: pop=%1 HR=%2 cash=%3", _popFIA, _hrCap, _cashCap];
+
+if(_hr + _hrDelta < _hrCap) then {
+	private _newHr = _hr + _hrDelta;
+	diag_log format ["Greenfork: Caps: HR: not capping %1 -> %2", _hr, _newHr];
+	_hr = _newHr;
+} else {
+	private _difference = _hr + _hrDelta - _hrCap;
+
+	if(_difference > 5) then {
+		private _newHr = round ((_hr + _hrCap) / 2);
+		diag_log format ["Greenfork: Caps: HR: drain down %1 -> %2", _hr, _newHr];
+		_hr = _newHr;
+	} else {
+		diag_log format ["Greenfork: Caps: HR: cap %1 -> %2", _hr, _hrCap];
+		_hr = _hrCap;
+	};
+};
+
+if(_resourcesFIA + _cashDelta < _cashCap) then {
+	private _newCash = round (_resourcesFIA + _cashDelta);
+	diag_log format ["Greenfork: Caps: Cash: not capping %1 -> %2", _resourcesFIA, _newCash];
+	_resourcesFIA = _newCash;
+} else {
+	private _difference = _resourcesFIA + _cashDelta - _cashCap;
+
+	if(_difference > 5000) then {
+		private _newCash = round ((_resourcesFIA + _cashCap) / 2);
+		diag_log format ["Greenfork: Caps: Cash: drain down %1 -> %2", _resourcesFIA, _newCash];
+		_resourcesFIA = _newCash;
+	} else {
+		diag_log format ["Greenfork: Caps: Cash: cap %1 -> %2", _resourcesFIA, _cashCap];
+		_resourcesFIA = _cashCap;
+	};
+};
+
+
+if (_hr < 0) then {_hr = 0};
+if (_resourcesFIA < 0) then {_resourcesFIA = 0};
+
+server setVariable ["hr", _hr, true];
+server setVariable ["resourcesFIA", _resourcesFIA, true];
Index: A3-Antistasi-Greenfork/src/description.ext
===================================================================
--- A3-Antistasi-Greenfork.orig/src/description.ext	2020-04-14 22:21:48.429361214 +0000
+++ A3-Antistasi-Greenfork/src/description.ext	2020-04-14 22:21:48.526360748 +0000
@@ -184,6 +184,12 @@
         texts[] = {"No","Yes"};
         default = 1;
     };
+    class minutesPerIncomeTick
+    {
+        title = "Minutes per income period";
+        values[] = {10, 15, 20, 30, 45, 60};
+        default = 20;
+    };
 };
 
 class CfgIdentities
Index: A3-Antistasi-Greenfork/src/initServer.sqf
===================================================================
--- A3-Antistasi-Greenfork.orig/src/initServer.sqf	2020-04-14 22:21:48.430361209 +0000
+++ A3-Antistasi-Greenfork/src/initServer.sqf	2020-04-14 22:21:48.527360743 +0000
@@ -29,6 +29,7 @@
 civTraffic = (paramsArray select 11) / 4; publicVariable "civTraffic";
 memberDistance = paramsArray select 13; publicVariable "memberDistance";
 limitedFT = if (paramsArray select 14 == 1) then {true} else {false}; publicVariable "limitedFT";
+incomeTick = (paramsArray select 15) * 60; publicVariable "incomeTick";
 _nul = call compile preprocessFileLineNumbers "initVar.sqf";
 initVar = true; publicVariable "initVar";
 savingServer = true;
