Index: A3-Antistasi-Greenfork/src/functions.hpp
===================================================================
--- A3-Antistasi-Greenfork.orig/src/functions.hpp	2020-06-07 15:01:18.783811275 +0000
+++ A3-Antistasi-Greenfork/src/functions.hpp	2020-06-07 15:01:23.538797030 +0000
@@ -258,5 +258,6 @@
         class resourceTaxes {file="resourceTaxes.sqf";};
         class sleepUntil {file="sleepUntil.sqf";};
         class isMilitaryBuilding {file="isMilitaryBuilding.sqf";};
+        class refreshFlyActions {file="refreshFlyActions.sqf";};
 	}
 }
Index: A3-Antistasi-Greenfork/src/initPlayerLocal.sqf
===================================================================
--- A3-Antistasi-Greenfork.orig/src/initPlayerLocal.sqf	2020-04-21 22:53:51.442391873 +0000
+++ A3-Antistasi-Greenfork/src/initPlayerLocal.sqf	2020-04-21 22:53:51.504391693 +0000
@@ -660,10 +660,11 @@
 petros disableAI "AUTOTARGET";
 petros addAction ["Mission Request", {nul=CreateDialog "mission_menu";},nil,0,false,true,"","_this == theBoss"];
 
-["itemAdd", ["A3A_ACECargoPlayerInit", {
+["itemAdd", ["A3A_PlayerGroupActions", {
 	[] spawn {
 		{
 			[_x] call A3A_fnc_refreshCargoActions;
+			[_x] call A3A_fnc_refreshFlyActions;
 		} forEach ((units player) - [player]);
 	};
 }, 15]] call BIS_fnc_loop;
Index: A3-Antistasi-Greenfork/src/refreshFlyActions.sqf
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ A3-Antistasi-Greenfork/src/refreshFlyActions.sqf	2020-04-21 20:40:13.481462404 +0000
@@ -0,0 +1,35 @@
+params ["_unit"];
+
+if (isNull _unit) exitWith {
+  diag_log format ["Greenfork: refreshFlyOrders: _unit is null"];
+  false;
+};
+
+private _flyInHeightActions = _unit getVariable ["A3A_FlyInHeightActions", []];
+
+{
+  private _actionId = _x select 0;
+  _unit removeAction _actionId;
+} forEach _flyInHeightActions;
+
+_flyInHeightActions = [];
+
+{
+  private _actionId = _unit addAction [
+    format ["Fly at %1m AGL", _x],
+    {
+      params ["_target", "_caller", "_actionId", "_args"];
+      (vehicle _target) flyInHeight (_args select 0);
+    },
+    [_x],
+    3,
+    false,
+    false,
+    "",
+    "vehicle _target isKindOf ""Air""",
+    0
+  ];
+  _flyInHeightActions pushBack [_actionId];
+} forEach [0, 15, 50, 250, 600, 1200, 2500, 8000];
+
+_unit setVariable ["A3A_FlyInHeightActions", _flyInHeightActions];
\ No newline at end of file
