---
 src/Missions/DES_Heli.sqf |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/src/Missions/DES_Heli.sqf
+++ b/src/Missions/DES_Heli.sqf
@@ -23,7 +23,7 @@
 while {true} do
 	{
 	_poscrashOrig = _posicion getPos [_dist,_ang];
-	if ((!surfaceIsWater _poscrash) and (_poscrash distance _posHQ < 4000)) exitWith {};
+	if ((!surfaceIsWater _poscrashOrig) and (_poscrashOrig distance _posHQ < 4000)) exitWith {};
 	_ang = _ang + 1;
 	_cuenta = _cuenta + 1;
 	if (_cuenta > 360) then
@@ -35,7 +35,7 @@
 
 _tipoVeh = selectRandom (vehPlanes + vehAttackHelis + vehTransportAir);
 
-_posCrashMrk = [_poscrash,random 500,random 360] call BIS_fnc_relPos;
+_posCrashMrk = [_poscrashOrig,random 500,random 360] call BIS_fnc_relPos;
 _posCrash = _posCrashOrig findEmptyPosition [0,100,_tipoVeh];
 if (count _posCrash == 0) then
 	{
