Index: A3-Antistasi-Greenfork/src/description.ext
===================================================================
--- A3-Antistasi-Greenfork.orig/src/description.ext	2020-04-21 20:50:48.417616191 +0000
+++ A3-Antistasi-Greenfork/src/description.ext	2020-04-21 20:52:28.487325226 +0000
@@ -2,10 +2,10 @@
 #include "dialogs.hpp"
 
 __EXEC(upstreamVersion = "1.5.0");
-__EXEC(downstreamVersion = "2020-03-01");
+__EXEC(downstreamVersion = "2020-04-21");
 
 author="Barbolani";
-OnLoadName = __EVAL(format ["Antistasi Altis %1 Greenfork %2", upstreamVersion, downstreamVersion]);
+OnLoadName = __EVAL(format ["A3-AAGF %1 %2", upstreamVersion, downstreamVersion]);
 OnLoadMission = "Be a Resistance leader, be a hero";
 loadScreen = "pic.JPG";
 
Index: A3-Antistasi-Greenfork/src/mission.sqm
===================================================================
--- A3-Antistasi-Greenfork.orig/src/mission.sqm	2020-04-21 20:46:08.465430217 +0000
+++ A3-Antistasi-Greenfork/src/mission.sqm	2020-04-21 20:51:52.961428523 +0000
@@ -153,7 +153,7 @@
 {
 	class Intel
 	{
-		briefingName="A3-Antistasi Altis";
+		briefingName="A3-Antistasi Altis 1.5.0 2020-04-21";
 		overviewText="Build FIA Army from scratch and defeat the AAF and CSAT forces in a whole map Dynamic Mission.";
 		resistanceWest=0;
 		startWeather=0;
