Index: A3-Antistasi-Greenfork/src/description.ext
===================================================================
--- A3-Antistasi-Greenfork.orig/src/description.ext	2020-04-04 18:22:03.344328749 +0000
+++ A3-Antistasi-Greenfork/src/description.ext	2020-04-04 19:26:34.912979034 +0000
@@ -2,7 +2,7 @@
 #include "dialogs.hpp"
 
 __EXEC(upstreamVersion = "1.5.0");
-__EXEC(downstreamVersion = "2020-03-01");
+__EXEC(downstreamVersion = "2020-04-14");
 
 author="Barbolani";
 OnLoadName = __EVAL(format ["Antistasi Altis %1 Greenfork %2", upstreamVersion, downstreamVersion]);
